test/lib/Benchmark.js

```diff
@@ -0,0 +1,259 @@
+/* globals XPCOMUtils, Services, Task */
+"use strict";
+
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "Task",
+                                  "resource://gre/modules/Task.jsm");
+
+// The default benchmark time is 1 second, i.e. 1e6 microseconds
+const DEFAULT_BENCH_TIME = 1e6;
+
+// For the performance.now(), so the timer resolution is 1 microsecond
+const performance = Services.appShell.hiddenDOMWindow.performance;
+
+// The max iteration number is 1e6 as the timer resolution is 1 microsecond
+const MAX_ITERATION = 1e6;
+
+/*
+ * Get the DOMHighResTimeStamp in microsecond
+ * See more detail at https://developer.mozilla.org/en-US/docs/Web/API/Performance/now
+ */
```

Comment:
Maybe in microseconds.  I think this comment needs to at least mention some of the bits/links from https://github.com/mozilla/activity-stream/pull/1209#issuecomment-244446467, and we probably still want to address that, even if it's by opening another ticket to do that work.